### ISSUE TITLE
Auto-download new songs from lists already set to download when connected to wifi

### ISSUE THREAD

**Issue by jonnyswan01** [UnicornsOnLSD/finamp#38]
not sure if this was supposed to work this way, but your download feature reminds a lot of what google play music did where you would set a playlist to download and then when you added a song to the playlist, that song would automatically download once you hit wifi. So, if you wanted it to work this, it currently does not. The new song just sits there, not downloaded while all the others are. Thanks!

**Comment by perniczech** [comment]
Yes, please! Would likely make this app **the** Play Music replacement 🥳

**Comment by drazil100** [comment]
I agree with the idea of having syncing instead of just straight up downloading. As I mentioned in #76 there are some issues with the current implimentation in that it assumes that the contents of the albums and playlists on the server don't change. I think that given that the content available is based on a server that if the server contents change it should update on the device (with the option to wait till you are on wifi)

**Comment by Swizzler121** [comment]
Also ran into this issue, it's super annoying that to update a downloaded playlist you have to delete the entire playlist from your device, then re-download the entire playlist. The ability to just sync over the added/removed songs would be great.

**Comment by UnicornsOnLSD** [comment]
An update button would be pretty easy to implement, I'll maybe fit it in the next minor release if I make one :)

**Comment by Fale** [comment]
This can be part of #213, I think

**Comment by Chaphasilor** [comment]
@jonnyswan01 the download system has been improved as part of the redesign beta. It will now sync all downloaded collections and songs when you start it while in online mode. There's also an option to only download when connected to WiFi.  
You can check the beta out here: https://github.com/jmshrv/finamp/releases/tag/0.9.2-beta

Do you have any more thoughts on when Finamp should synchronize with the server? Every time WiFi is reconnected seems excessive, and I'm not sure Android or iOS will let us do that. There is a dedicated sync button on the downloads page, but maybe some of you have an even better solution?

**Comment by Chaphasilor** [comment]
Adding a song to a playlist will now (in the beta) also automatically download that song if the playlist was already downloaded. If you have Finamp set to only download over WiFi, it will of course not download while you're on mobile data. The download would then be queued and automatically started as soon as you're connected to WiFi.

I'll close this issue for now, let me know if anything isn't working in the beta!
